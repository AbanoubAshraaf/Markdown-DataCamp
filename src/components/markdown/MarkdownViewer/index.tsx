import React from 'react';
import { ScrollView, StyleSheet } from 'react-native';
import Markdown from 'react-native-markdown-display';

interface MarkdownViewerProps {
  markdownText: string;
}

const MarkdownViewer: React.FC<MarkdownViewerProps> = ({ markdownText }) => (
  <ScrollView style={styles.resultArea}>
    <Markdown>{markdownText}</Markdown>
  </ScrollView>
);

const styles = StyleSheet.create({
  resultArea: {
    flex: 1,
    padding: 10,
    margin:10,
    backgroundColor: '#f0f0f0',
    borderRadius: 5,
  },
});

export default MarkdownViewer;
