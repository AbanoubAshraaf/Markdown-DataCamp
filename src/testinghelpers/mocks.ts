export const mockedMarkDownText = 'markdown text'

export const mockedPortraitDimensions = {
    width: 400,
    height: 800,
    scale: 0,
    fontScale: 0,
};

export const mockedLandScapeDimensions = {
    width: 800,
    height: 400,
    scale: 0,
    fontScale: 0,
};