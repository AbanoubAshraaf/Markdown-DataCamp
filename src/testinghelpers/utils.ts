import { RenderResult } from "@testing-library/react-native";

interface Screen extends RenderResult {
    isDetached?: boolean;
}

export const createSnapShot = (screen: Screen) => {
    expect(screen).toMatchSnapshot();
  };