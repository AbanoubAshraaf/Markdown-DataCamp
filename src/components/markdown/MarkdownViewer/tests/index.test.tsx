import React from 'react';
import { render, screen } from '@testing-library/react-native';
import MarkdownViewer from '..';
import { createSnapShot } from 'src/testinghelpers/utils';

describe('MarkdownViewer', () => {
  it('renders markdown text correctly', () => {
    const markdownText = '# Hello World';
    const { getByText } = render(<MarkdownViewer markdownText={markdownText} />);

    expect(getByText('Hello World')).toBeTruthy();
    createSnapShot(screen)
  });
});
