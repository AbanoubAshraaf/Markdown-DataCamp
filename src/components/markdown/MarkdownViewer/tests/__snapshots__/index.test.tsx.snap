// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownViewer renders markdown text correctly 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "#f0f0f0",
      "borderRadius": 5,
      "flex": 1,
      "margin": 10,
      "padding": 10,
    }
  }
>
  <View>
    <View
      style={{}}
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={{}}
        >
          <Text
            style={
              [
                {
                  "fontSize": 32,
                },
                {},
              ]
            }
          >
            Hello World
          </Text>
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
